### DISCUSSION TITLE
Argument Error in randomBN

### DISCUSSION THREAD

**Discussion by kenneth-lee-ch** [agrumery/aGrUM#11] [Q&A] [answered]
How come the `randomBN` function throws an error when the `n` argument is small? I notice `n` needs to be at least 5. 

```
gum.randomBN(n= 4, domain_size= 2, ratio_arc = 1)
```
![Screenshot 2023-04-07 at 2 20 32 PM](https://user-images.githubusercontent.com/24616330/230658398-1b7be097-65e8-4a0e-9279-7708cd86e793.png)

**Comment by phwuil** [comment, marked as answer]
The way BN are randomly generated using a MCMC approach makes it optimized in the sens that the distribution over all structure is (close to) uniform, even if there are constraints on the structure. However, this make it difficult and then very unoptimized in terms of time generation for very small graphs. Moreover, the  `ratio_arc` argument is not very relevant for small graphs (with 3 nodes, the number of arcs for a connected graph is between 2 and 3, so the relevant ratio would be only 1 and 1.5).

However, I understand that it may seem strange and propose to accept 4 nodes (but not 3 :-) ).

**Comment by phwuil** [comment]
![image](https://user-images.githubusercontent.com/5980636/230783607-d5067f6d-5948-48f9-b62a-aaacb254056c.png)
